charts/argo-applicationset/templates/deployment.yaml

```diff
@@ -0,0 +1,59 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "argo-applicationset.fullname" . }}
+  labels:
+    {{- include "argo-applicationset.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
```

Comment:
if replicaCount > 1 then we should also do `enable-leader-election=true`

Reply:
Done, I used the solution suggested here https://github.com/helm/helm/issues/3005#issuecomment-501471161